allennlp/commands/train.py

```diff
@@ -203,16 +174,55 @@ def train_model(params: Params, serialization_dir: str, file_friendly_logging: b
 
     evaluate_on_test = params.pop_bool("evaluate_on_test", False)
     params.assert_empty('base train command')
-    trainer.train()
+    metrics = trainer.train()
 
     # Now tar up results
     archive_model(serialization_dir, files_to_archive=params.files_to_archive)
 
     if test_data and evaluate_on_test:
-        evaluate(model, test_data, iterator, cuda_device=trainer._cuda_device)  # pylint: disable=protected-access
+        test_metrics = evaluate(model, test_data, iterator, cuda_device=trainer._cuda_device)  # pylint: disable=protected-access
+        for k, v in test_metrics.items():
+            metrics["test_" + k] = v
 
     elif test_data:
         logger.info("To evaluate on the test set after training, pass the "
                     "'evaluate_on_test' flag, or use the 'allennlp evaluate' command.")
 
+    metrics_json = json.dumps(metrics, indent=2)
+    with open(os.path.join(serialization_dir, "metrics.json"), "w") as f:
+        f.write(metrics_json)
+    logger.info("Metrics: " + metrics_json)
```

Comment:
I think it's helpful logging output to have all of your stats displayed at the end, but if anyone doesn't want this output here I'll remove it.